solr/core/src/test/org/apache/solr/core/ExitableDirectoryReaderTest.java

```diff
@@ -70,9 +70,12 @@ public void testPrefixQuery() throws Exception {
 
     // this time we should get a query cache hit and hopefully no exception?  this may change in the
     // future if time checks are put into other places.
-    assertJQ(req("q", q, "timeAllowed", "1", "sleep", sleep), assertionString);
+    // 2024-4-15: it did change..., and now this fails with 1 or 2 ms and passes with 3ms... I see
+    // no way this won't be terribly brittle. Maybe TestInjection of some sort to bring this back?
```

Comment:
Maybe use `CallerSpecificQueryLimit` (example in `TestQueryLimits.testQueryLimits()`.

Reply:
I don't think that helps. The test wants to verify that the result was cached and is hoping that the request **_won't_** time out because of that, but that only worked because timeAllowed was only measured/triggered in a small area of code sensitive to the availability of the cache. Now we check it all over the place. If we raise the limit too much it's not a test, and if this test gets starved for resources it will fail spuriously.

Reply:
What we really need for this test is a cache admin command that can list the contents of the a cache (and one would expect a manual clear capability too, though that's not needed here) One might be able to get by with a metrics call, looking for an increment in cache hits, but that's pretty heavy reliance on coincidence rather than explicit testing of the actual desired behavior.

Reply:
I just looked at the ability to check directly. There is some hope since we could 
```
    SolrCores solrCores = ExitableDirectoryReaderTest.h.getCoreContainer().solrCores;
    List<SolrCore> cores = solrCores.getCores();
    for (SolrCore core : cores) {
      // identify core we need here...
    }
```

but assuming we can identify the appropriate core, there isn't even an internal api to check caches. Would look for something steming from `core.getSearcher().get().???` but we would need to add an accessor for the queryCache  and then on SolrCache a way to inspect the keys of the cache... that seems way beyond what this change should attempt.